src/posthog-core.js

```diff
@@ -339,9 +340,17 @@ PostHogLib.prototype._handle_queued_event = function (url, data, options) {
 }
 
 PostHogLib.prototype.__compress_and_send_json_request = function (url, jsonData, options, callback) {
-    if (this.get_config('_capture_metrics')) {
-        this._send_request(url, jsonData, { ...options, plainJSON: true }, callback)
-    } else if (this.compression['lz64'] || (options.compression && options.compression === 'lz64')) {
+    if (this.compression['gzip-js']) {
+        // :TRICKY: This is encoded as latin-1/iso-8859-1, needs to be decoded accordingly on the server.
+        const compressed = strFromU8(gzipSync(strToU8(jsonData)), true)
+
+        this._send_request(
+            url,
+            compressed,
```

Comment:
Curious on the rationale for passing data this way instead of the same way we do it for LZ64 compression (`{ data: LZString.compressToBase64(jsonData), compression: 'lz64' }`), if this is because the data is binary, then we could have some unhandled edge cases (e.g. if `config.test = True`, see https://github.com/PostHog/posthog-js/blob/20ee66b81ae0902ffbe3bf17dad3a4a502266076/src/posthog-core.js#L384)

Reply:
Great comments!

The lines you mentioned lines are (effectively) dead, but require further refactoring to make it so in practice. I will do so soon though.

The reason we're passing data this way has to do with the request itself:
1. We can do a Content-Type: application/x-www-form-urlencoded request (as with lzstring) - this requires base64-encoding the data, which ends up wasting bytes
2. Or we can do a Content-Type: text/plain (as we do here) request - this doesn't require base64-encoding the data if we can get both of the ends to encode/decode the data correctly.

Normally 1. is simpler but takes more space - however I couldn't get base64-encoded gzip working without doing significant js-side changes so decided to instead extend the posthog server.

Reply:
That makes total sense! Seems like option 2 might be best in the long run